workflow/applications/applications.py

```diff
@@ -31,94 +30,84 @@ def __init__(self, conf: Configuration) -> None:
-
-        self.nens = base.get('NMEM_ENS', 0)
-        self.fcst_segments = base.get('FCST_SEGMENTS', None)
-        self.interval_gfs = to_timedelta(f"{base.get('INTERVAL_GFS')}H")
-
-        if not AppConfig.is_monotonic(self.fcst_segments):
-            raise ValueError(f'Forecast segments do not increase monotonically: {",".join(self.fcst_segments)}')
-
-        self.wave_runs = None
-        if self.do_wave:
-            wave_run = base.get('WAVE_RUN', 'BOTH').lower()
-            if wave_run in ['both']:
-                self.wave_runs = ['gfs', 'gdas']
-            elif wave_run in ['gfs', 'gdas']:
-                self.wave_runs = [wave_run]
-
-        self.aero_anl_runs = None
-        self.aero_fcst_runs = None
-        if self.do_aero:
-            aero_anl_run = base.get('AERO_ANL_RUN', 'BOTH').lower()
-            if aero_anl_run in ['both']:
-                self.aero_anl_runs = ['gfs', 'gdas']
-            elif aero_anl_run in ['gfs', 'gdas']:
-                self.aero_anl_runs = [aero_anl_run]
-            aero_fcst_run = base.get('AERO_FCST_RUN', None).lower()
-            if aero_fcst_run in ['both']:
-                self.aero_fcst_runs = ['gfs', 'gdas']
-            elif aero_fcst_run in ['gfs', 'gdas']:
-                self.aero_fcst_runs = [aero_fcst_run]
+        print(f"Generating the XML for a {self.mode}_{self.net} case")
 
     def _init_finalize(self, conf: Configuration):
-        print("Finalizing initialize")
-
-        # Get a list of all possible config_files that would be part of the application
-        self.configs_names = self._get_app_configs()
-
-        # Source the config files for the jobs in the application without specifying a RUN
-        self.configs = {'_no_run': self._source_configs(conf)}
-
-        # Update the base config dictionary based on application
-        self.configs['_no_run']['base'] = self._update_base(self.configs['_no_run']['base'])
+        '''
+        Finalize object initialization calling subclass methods
+        '''
 
-        # Save base in the internal state since it is often needed
-        base = self.configs['_no_run']['base']
+        # Get run-, net-, and mode-based options
+        self.run_options = self._get_run_options(conf)
 
-        # Get task names for the application
+        # Get task names and runs for the application
         self.task_names = self.get_task_names()
 
-        # Finally, source the configuration files for each valid `RUN`
-        for run in self.task_names.keys():
+        # Initialize the configs and model_apps dictionaries
+        self.configs = dict.fromkeys(self.runs)
+
+        # Now configure the experiment for each valid run
+        for run in self.runs:
             self.configs[run] = self._source_configs(conf, run=run, log=False)
 
-            # Update the base config dictionary based on application and RUN
-            self.configs[run]['base'] = self._update_base(self.configs[run]['base'])
+    @abstractmethod
```

Comment:
This method looks pretty concrete to me.
```suggestion
```

Reply:
You're right and I learned something new.  Thanks!